File: app/src/main/java/command/ceiba/com/co/dojocommandpattern/BaseActivityCommand.java
package command.ceiba.com.co.dojocommandpattern;

import android.content.Context;

import java.util.Map;

public abstract class BaseActivityCommand {

    protected Context context;
    protected NotificationReceiver receiver;
    protected Map<String, String> notification;


    protected BaseActivityCommand(Context context, Map<String, String> notification){
        this.context = context;
        this.notification = notification;

        receiver = new NotificationReceiver(context);
    }
}

File: app/src/main/java/command/ceiba/com/co/dojocommandpattern/BaseActivityCommand.kt
package command.ceiba.com.co.dojocommandpattern

import android.content.Context

abstract class BaseActivityCommand protected constructor(
    protected var context: Context,
    protected var notification: Map<String, String>
) {
    protected var receiver: NotificationReceiver

    init {
        receiver = NotificationReceiver(context)
    }
}